### ISSUE TITLE
Object only used in String template is marked as unused

### ISSUE THREAD

**Issue by EotT123** [manifold-systems/manifold-ij#39]
Objects that are only used in a String Template are marked as unused.

![screenshot 47](https://github.com/manifold-systems/manifold-ij/assets/32986779/174c5357-edb2-4a4a-9eb2-69c565888c88)
![screenshot 48](https://github.com/manifold-systems/manifold-ij/assets/32986779/5816c77d-757f-4011-a032-69f619369747)

Also applies to method parameters:

![screenshot 50](https://github.com/manifold-systems/manifold-ij/assets/32986779/c8796e23-9ca0-4f14-af70-9b4d60bdcabd)

**Comment by EotT123** [comment]
An even worse case:  
Using a static method call inside a String template expression, e.g.
```java
String text = "${StringUtils.substring(value, 1)}"
```
If `StringUtils` class is not used elsewhere, it is marked as unused. If configured in IntelliJ, it gets removed automatically from the list of imports, resulting in a compilation failure.

**Comment by rsmckinney** [comment]
I am not seeing any of this, all of this is implemented.
![image](https://github.com/manifold-systems/manifold-ij/assets/588843/b49540ad-1b8d-41bc-b0d1-a2949b73e6d1)
What version of the plugin / IJ / etc. are you using?

**Comment by rsmckinney** [comment]
I am not seeing any of this, all of this is implemented.
![image](https://github.com/manifold-systems/manifold-ij/assets/588843/b49540ad-1b8d-41bc-b0d1-a2949b73e6d1)
What version of the plugin / IJ / etc. are you using?

**Comment by EotT123** [comment]
Latest version of everything:
* Manifold plugin: 2024.1.5
* IntelliJ IDEA 2024.1.2 (Community Edition)

This also happens for me with the [sample project](https://github.com/manifold-systems/manifold/blob/master/manifold-deps-parent/manifold-strings-test/src/test/java/manifold/strings/StringTemplateTest.java):

![screenshot 51](https://github.com/manifold-systems/manifold-ij/assets/32986779/aa47aef7-35c1-477f-ae03-ea99dc374248)

**Comment by rsmckinney** [comment]
Looking at your earlier screenshots, it appears string templates aren't really enabled, for instance local var usage s/b highlighted. This is probably the root of the problem. Can you provide your build file[s]? A simple change will probably fix things up for you.

**Comment by EotT123** [comment]
It is enabled. Using the manifold-strings-test project (I didn't change anything), all tests succeeds, but the 'never used' warning is still present.
![screenshot 48](https://github.com/manifold-systems/manifold-ij/assets/32986779/b26c5f23-c947-4029-a565-74595d3379f8)

**Comment by rsmckinney** [comment]
Yeah, most of the test modules aren't setup for IntelliJ use. Always refer to the setup docs for manifold features. In this case, see the [manifold-strings setup docs](https://github.com/manifold-systems/manifold/tree/master/manifold-deps-parent/manifold-strings#maven). Also see the [android docs](http://manifold.systems/android.html) if you're working in that environment.

**Comment by superxiao** [comment]
I have the same issue. Intellij thinks a static function that's used in a string template as unused, and removes it when organizing imports. Intellij manifold plugin version 2024.1.6, pom.xml:

```
  <properties>
    <manifold.version>2024.1.21</manifold.version>
  </properties>
  <dependencies>
    <dependency>
      <groupId>systems.manifold</groupId>
      <artifactId>manifold-ext-rt</artifactId>
      <version>${manifold.version}</version>
    </dependency>
    <dependency>
      <groupId>systems.manifold</groupId>
      <artifactId>manifold-rt</artifactId>
      <version>${manifold.version}</version>
    </dependency>
    <dependency>
      <groupId>systems.manifold</groupId>
      <artifactId>manifold-collections</artifactId>
      <version>${manifold.version}</version>
    </dependency>
    <dependency>
      <groupId>systems.manifold</groupId>
      <artifactId>manifold-strings</artifactId>
      <version>${manifold.version}</version>
    </dependency>
  </dependencies>
  <build>
    <plugins>
      <plugin>
        <groupId>org.apache.maven.plugins</groupId>
        <artifactId>maven-compiler-plugin</artifactId>
        <version>3.11.0</version>
        <configuration>
          <source>17</source>
          <target>17</target>
          <encoding>UTF-8</encoding>
          <compilerArgs>
            <!-- Configure manifold plugin -->
            <arg>-Xplugin:Manifold</arg>
          </compilerArgs>
          <annotationProcessorPaths>
            <path>
              <groupId>systems.manifold</groupId>
              <artifactId>manifold</artifactId>
              <version>${manifold.version}</version>
            </path>
            <path>
              <groupId>systems.manifold</groupId>
              <artifactId>manifold-ext</artifactId>
              <version>${manifold.version}</version>
            </path>
            <path>
              <groupId>systems.manifold</groupId>
              <artifactId>manifold-strings</artifactId>
              <version>${manifold.version}</version>
            </path>
            <path>
              <groupId>systems.manifold</groupId>
              <artifactId>manifold-exceptions</artifactId>
              <version>${manifold.version}</version>
            </path>
          </annotationProcessorPaths>
        </configuration>
      </plugin>
    </plugins>
  </build>
```
